api/pacman-api-compliance/src/main/java/com/tmobile/pacman/api/compliance/repository/ComplianceRepositoryImpl.java

```diff
@@ -2342,9 +2344,13 @@ public IssueExceptionResponse exemptAndUpdateMultipleIssueDetails(String assetGr
                         String _index = dataSource;
                         String _type = targetType + "_audit";
                         if(!skipAuditTrail) {
-                            Map<String, String> optionalAuditFields = new HashMap<>(2);
-                            optionalAuditFields.put(Constants.ISSUE_EXEMPTION_EXPIRY_DATE, sdf.format(issuesException.getExceptionEndDate()));
-                            optionalAuditFields.put(Constants.ISSUE_EXEMPTION_REASON, issuesException.getExceptionReason());
+                            Map<String, String> optionalAuditFields = Collections.emptyMap();
+                            if (issuesException.getExceptionEndDate() != null) {
+                                optionalAuditFields.put(Constants.ISSUE_EXEMPTION_EXPIRY_DATE, sdf.format(issuesException.getExceptionEndDate()));

```

Comment:
If you omit this when it comes empty, what is the consequence? Who is sending this field? If it's the UI, then let's see why it comes empty.

Reply:
@ershad-paladin - was thinking about this over night and here is what I believe is the better approach as this seems more of a hack then a permanent fix.

Let's do a minor refactor to separate (admin vs user functionalities) or (request vs revoke, deny).

1. This is a much cleaner a more robust Admin functionality
2. Avoids confusing check and lets the next dev know exactly what the request for Admin vs User functionality is
3. You can easily abstract this to some base user model class and apply some inheritance.